R/npn_data_download.R

```diff
@@ -743,46 +747,40 @@ npn_get_data_by_year <- function(endpoint,
 #' @keywords internal
 #' @examples \dontrun{
 #' npn_get_data(
-#'   url = "https://services.usanpn.org/npn_portal//observations/getObservations.ndjson?",
+#'   endpoint = "observations/getObservations.ndjson",
 #'   query = list(
-#'     request_src = "Unit%20Test",
+#'     request_src = "Unit Test",
 #'     climate_data = "0",
 #'     `species_id[1]` = "6",
 #'     start_date = "2010-01-01",
 #'     end_date = "2010-12-31"
 #'   )
 #' )
 #' }
-npn_get_data <- function(url,
+npn_get_data <- function(endpoint,
                          query,
                          download_path = NULL,
                          always_append = FALSE,
                          six_leaf_raster = NULL,
                          six_bloom_raster = NULL,
                          agdd_layer = NULL,
                          additional_layers = NULL) {
-  req <- httr2::request(url) %>%
-    httr2::req_user_agent("rnpn (https://github.com/usa-npn/rnpn/)") %>%
-    httr2::req_method("POST") %>%
-    httr2::req_body_form(!!!query)
 
-  con <- httr2::req_perform_connection(req)
-  on.exit(close(con), add = TRUE)
+  if (is.null(download_path)) { #use JSON
+    endpoint <- sub("(?<=\\.)\\w+$", "json", endpoint, perl = TRUE)
+  } else { #use NDJSON
+    endpoint <- sub("(?<=\\.)\\w+$", "ndjson", endpoint, perl = TRUE)
+  }
 
-  dtm <- tibble::tibble()
-  i <- 0
-  while (!httr2::resp_stream_is_complete(con)) {
-    resp <- httr2::resp_stream_lines(con, lines = 5000)
+  req <- base_req %>%
+    httr2::req_url_path_append(endpoint) %>%
+    httr2::req_progress() %>%
+    httr2::req_method("POST") %>%
+    httr2::req_body_form(!!!query)
```

Comment:
@jeffswitzer, is there any reason to include the query as the body in a POST request vs. just having it part of the URL in a regular GET request?